qt/widgets/mplcpp/test/PlotTest.h

```diff
@@ -0,0 +1,157 @@
+  //     Python::NewRef(PyImport_Import(Py_BuildValue("s", "matplotlib")));
+  // auto args = Python::NewRef(Py_BuildValue("(s)", "Agg"));
+  // auto kwargs = Python::NewRef(Py_BuildValue("{}"));
+  // funcsModule.attr("use")(args, kwargs);
+  auto mpl = Python::NewRef(PyImport_ImportModule("matplotlib"));
+  mpl.attr("use")("Agg");
+}
+} // namespace
+
+class PlotTest : public CxxTest::TestSuite {
+public:
+  static PlotTest *createSuite() { return new PlotTest; }
+  static void destroySuite(PlotTest *suite) { delete suite; }
+
+  void setUp() override {
+    Mantid::API::FrameworkManager::Instance();
+    auto alg = Mantid::API::AlgorithmManager::Instance().createUnmanaged(
+        "CreateSampleWorkspace");
+    alg->initialize();
+    alg->setProperty("OutputWorkspace", "ws");
+    alg->execute();
+
+    setMatplotlibBackend();
+  }
+
+  void testPlottingWorksWithWorkspaceIndex() {
+    std::vector<std::string> workspaces = {"ws"};
+    std::vector<int> index = {1};
+    TS_ASSERT_THROWS_NOTHING(plot(workspaces, boost::none, index, boost::none,
+                                  boost::none, boost::none, boost::none))
+  }
+
+  void testPlottingWorksWithSpecNum() {
+    std::vector<std::string> workspaces = {"ws"};
+    std::vector<int> index = {1};
+    TS_ASSERT_THROWS_NOTHING(plot(workspaces, index, boost::none, boost::none,
+                                  boost::none, boost::none, boost::none))
+  }
+
+  void testPlottingThrowsWithSpecNumAndWorkspaceIndex() {
+    std::vector<std::string> workspaces = {"ws"};
+    std::vector<int> index = {1};
+    TS_ASSERT_THROWS(plot(workspaces, index, index, boost::none, boost::none,
+                          boost::none, boost::none),
+                     const std::invalid_argument &)
+  }
+
+  void testPlottingWithPlotKwargs() {
+    std::vector<std::string> workspaces = {"ws"};
+    std::vector<int> index = {1};
+    QHash<QString, QVariant> hash;
+    hash.insert(QString("linewidth"), QVariant(10));
+    TS_ASSERT_THROWS_NOTHING(plot(workspaces, index, boost::none, boost::none,
+                                  hash, boost::none, boost::none))
+  }
+
+  void testPlottingWithIncorrectPlotKwargsThrows() {
+    std::vector<std::string> workspaces = {"ws"};
+    std::vector<int> index = {1};
+    QHash<QString, QVariant> hash;
+    hash.insert(QString("asdasdasdasdasd"), QVariant(1));
+    TS_ASSERT_THROWS(plot(workspaces, index, boost::none, boost::none, hash,
+                          boost::none, boost::none),
+                     const PythonException &)
+  }
+
+  void testPlottingWithAxProperties() {
+    std::vector<std::string> workspaces = {"ws"};
+    std::vector<int> index = {1};
+    QHash<QString, QVariant> hash;
+    hash.insert(QString("xscale"), QVariant("log"));
+    TS_ASSERT_THROWS_NOTHING(plot(workspaces, index, boost::none, boost::none,
+                                  boost::none, hash, boost::none))
+  }
+
+  void testPlottingWithAxPropertiesThrows() {
```

Comment:
```suggestion
  void testPlottingWithUnknownAxPropertiesThrows() {
```

Reply:
I'll replace `Unknown` with Incorrect as that is used for the plotKwargs test.